src/screens/Question/index.tsx

```diff
@@ -1,87 +1,145 @@
 
 import { ReactComponent as ArrowRight } from 'assets/images/icons/arrow-right.svg';
 import { ReactComponent as CloseButton } from 'assets/images/icons/close-btn.svg';
 import Answer from 'components/Answer';
 import ElevatedButton from 'components/ElevatedButton';
+import LoadingDialog from 'components/LoadingDialog';
 import MainView from 'components/MainView';
-import { useAppSelector } from 'hooks';
+import { useAppDispatch, useAppSelector } from 'hooks';
+import { paths, questionCompletePath } from 'routes';
+import { submitSurveyAsyncThunk, surveyAction } from 'store/reducers/Survey';
+import { AnswerRequest } from 'types/request/surveySubmitRequest';
 
 export const questionScreenTestIds = {
   index: 'question__index',
   title: 'question__title',
   closeButton: 'question__close-button',
   nextButton: 'question__next-button',
+  submitButton: 'question__submit-button',
 };
 
 const QuestionScreen = (): JSX.Element => {
-  const { survey } = useAppSelector((state) => state.survey);
+  const { survey, questionRequests, isLoading, isError, isSubmitSuccess } = useAppSelector((state) => state.survey);
+  const dispatch = useAppDispatch();
   const navigate = useNavigate();
 
   const [currentQuestion, setCurrentQuestion] = useState(survey?.questions?.at(0));
   const [questionIndex, setQuestionIndex] = useState(0);
 
+  const handleAnswerChanged = (answers: AnswerRequest[]) => {
+    dispatch(
+      surveyAction.fillAnswers({
+        id: currentQuestion?.id ?? '',
+        answers: answers,
+      })
+    );
+  };
+
+  const isAnswerAdded = (): boolean => {
+    if (currentQuestion?.displayType === 'intro' || currentQuestion?.displayType === 'outro') {
+      return true;
+    }
+
+    return questionRequests.some((value) => value.id === currentQuestion?.id);
+  };
+
   const onNextClick = () => {
+    if (!isAnswerAdded()) {
+      toast.warn('Please answer the question first!', { position: 'top-center' });
+      return;
+    }
+
     const nextQuestionIndex = questionIndex + 1;
     if (nextQuestionIndex >= (survey?.questions?.length ?? 0)) {
       return;
     }
+
     setCurrentQuestion(survey?.questions?.at(nextQuestionIndex));
     setQuestionIndex(nextQuestionIndex);
   };
 
   const onSubmitClick = () => {
-    // TODO submit survey
+    dispatch(
+      submitSurveyAsyncThunk({
+        surveyId: survey?.id ?? '',
+        questions: questionRequests,
+      })
+    );
   };
 
-  function goBack() {
-    navigate(-1);
+  function handleOnClose() {
+    dispatch(surveyAction.resetState());
+    navigate(paths.root, { replace: true });
   }
 
+  useEffect(() => {
+    if (isSubmitSuccess) {
+      navigate(questionCompletePath());
+      dispatch(surveyAction.resetState());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isSubmitSuccess]);
+
+  useEffect(() => {
+    if (isError) {
+      toast.error('There is something wrong. Please try again later!', { position: 'top-center' });
```

Comment:
those toast seem generic, do you think it's worth creating the common toast for wide use?

Reply:
@nvminhtue updated in [7ac6657](https://github.com/manh-t/react-survey/pull/85/commits/7ac66571ea2cbc5bcf4d18543d30d95856c04a3c)